packages/gitbook/src/lib/visitors.ts

```diff
@@ -221,6 +228,28 @@ function parseVisitorQueryParamValue(value: string): ClaimPrimitive {
     return value;
 }
 
+/**
+ * Returns to cookie response to use in order to persist visitor params that were passed to the URL.
+ */
+export function getResponseCookieForVisitorParams(
+    visitorUnsignedClaims: VisitorPayloadLookup['unsignedClaims']
+): ResponseCookie | undefined {
+    if (Object.keys(visitorUnsignedClaims.fromVisitorParams).length === 0) {
+        return undefined;
+    }
+
+    return {
+        name: VISITOR_UNSIGNED_CLAIMS_PREFIX,
```

Comment:
Using the `gitbook-visitor-public` but happy to change to store the params in a separate/dedicated one.